### ISSUE TITLE
Eager loading occurs on association.create whereas it should be using a query proxy

### ISSUE THREAD

**Issue by rahulmeena13** [neo4jrb/neo4j#1216]
Hey, 
Say I have the following models:
```
class User
  include Neo4j::ActiveNode
  has_many :out, :sessions, rel_class: :UserSession
end
```
```
class Session
  include Neo4j::ActiveNode
  #For examples sake, normally there would be only one user associated with a session
  has_many :in, :users, rel_class: :UserSession 
end
```

And an ActiveRel model for UserSession:

```
class UserSession
  include Neo4j::ActiveRel
  from_class :User
  to_class :Session
  creates_unique

  type :HAS_SESSION
  property :device_type, type: String
end
```

When I loop over users and create new relationship like this:

User.all.each do |user|
   session = ...
   user.sessions.create(session, device_type: mobile)
end

The following query is generated:

`MATCH (previous:User) WHERE (ID(previous) IN {ID_previous}) OPTIONAL MATCH (previous)-[rel1:HAS_SESSION]->(next:Session) RETURN ID(previous), collect(next) | {:ID_previous=>[547738, 2915210, 2915211, 2915212, 2915213, 2915214, 2915215, 2915216, 2915217, 2915218, 2915219, 2915220, 2915221, 2915222, 2915223, 2915224, 2915225, 2915226, 2915227, 2915228, 2915229, 2915230, 2915231, 2915232, 2915233, 2915234, 2915235, 2915236, 2915237, 2915238, 2915239, 2915240, 2915241, 2915242, 2915243, 2915244, 2915245, 2915246, 2915247, 2915248, 2915249, 2915250, 2915251, 2915252, 2915253, 2915254, 2915255, 2915256, 2915257, 2915258, 2915259, 2915260, 2915261, 2915262, 2915263, 2915264, 2915265, 2915266, 2915267, 2915268, 2915269, 2915270, 2915271, 2915272…]}`

The ID_previous contains all the ids for the User label, so say if I have 50k users, this query will contain 50 thousand ids in ID_previous param.

Ideally there should be a query proxy for the same instead of eagerloading all users and their sessions.

Neo4j database version: 3.0.0
`neo4j` gem version: 7.0.4
`neo4j-core` gem version: 6.1.3

**Comment by cheerfulstoic** [comment]
Ok, I think I've got it.  It's running through Travis now, but could you try the following `Gemfile` line?

```ruby
gem 'neo4j', github: 'neo4jrb/neo4j', branch: 'fix_for_1216'
```

**Comment by rahulmeena13** [comment]
Just tried it. It's fixed! Verified the relation was created, and my script which usually took about 5 minutes to run only took 1 minute. Thanks!

**Comment by cheerfulstoic** [comment]
Great, thanks!  Just released the fix as 7.0.10.  Thanks for the report!
